File: TransCoder_tokenized_test_set_functions/python/MINIMUM_INSERTIONS_TO_FORM_A_PALINDROME_WITH_PERMUTATIONS_ALLOWED.py
def minInsertion ( tr1 ) :
    n = len ( str1 )
    res = 0
    count = [ 0 for i in range ( 26 ) ]
    for i in range ( n ) :
        count [ ord ( str1 [ i ] ) - ord ( 'a' ) ] += 1
    for i in range ( 26 ) :
        if ( count [ i ] % 2 == 1 ) :
            res += 1
    if ( res == 0 ) :
        return 0
    else :
        return res - 1
    

File: results/java/MINIMUM_INSERTIONS_TO_FORM_A_PALINDROME_WITH_PERMUTATIONS_ALLOWED.cpp
int f_filled ( string str ) {
  int n = str . length ( ) ;
  int res = 0 ;
  vector < int > count ( 26 , 0 ) ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    count [ str [ i ] - 'a' ] ++ ;
  }
  for ( int i = 0 ;
  i < 26 ;
  i ++ ) {
    if ( count [ i ] % 2 == 1 ) {
      res ++ ;
    }
  }
  return ( res == 0 ) ? 0 : res - 1 ;
}
